dali/dfu/dfuutil.cpp

```diff
@@ -858,9 +858,38 @@ class CDFUhelper: public CInterface, implements IDFUhelper
                 throwError1(DFUERR_DFileNotFound, lfn);
         }
         else {
-            Owned<IFileDescriptor> fdesc = file->getFileDescriptor();
-            Owned<IPropertyTree> t = fdesc->getFileTree();
-            toXML(t, out, true);
+            CDfsLogicalFileName tmplfn;
+            StringBuffer lfnpath;
+            bool safe = false;
+            if ((strstr(lfn, "::") != NULL) && !strchr(lfn, '/'))
```

Comment:
Yes, the IDistributedFileDirectory::getFileTree() can export the file xml. The result is fine for plain file, but a bit  strange with superfile:
```xml
 <File directory="/var/lib/HPCCSystems/hpcc-data/thor" numclusters="1" numparts="6">
  <Attr fileCrc="-570320118"
        footerLength="0"
        headerLength="0"
        owner=""/>
  <Cluster defaultBaseDir="/var/lib/HPCCSystems/hpcc-data/thor"
           defaultReplicateDir="/var/lib/HPCCSystems/hpcc-mirror/thor"
           interleave="2"
           mapFlags="48">
   <Group>192.168.1.56</Group>
  </Cluster>
  <Part fileCrc="-1256801527"
        modified="2015-09-11T15:58:23"
        name="book2.csv._1_of_3"
        num="1"
        size="34"/>
  <Part fileCrc="0"
        modified="2015-09-09T14:36:24"
        name="book1.csv._1_of_3"
        num="2"
        size="0"/>
  <Part fileCrc="80761555"
        modified="2015-09-11T15:58:23"
        name="book2.csv._2_of_3"
        num="3"
        size="34"/>
  <Part fileCrc="0"
        modified="2015-09-09T14:36:24"
        name="book1.csv._2_of_3"
        num="4"
        size="0"/>
  <Part fileCrc="1051947210"
        modified="2015-09-11T15:58:23"
        name="book2.csv._3_of_3"
        num="5"
        size="34"/>
  <Part fileCrc="-570320118"
        modified="2015-09-09T14:36:24"
        name="book1.csv._3_of_3"
        num="6"
        size="102"/>
 </File>
 <SuperFile interleaved="2"
            modified="2015-09-14T09:08:05"
            name="superfile"
            numsubfiles="2">
  <SubFile name=".::book2.csv" num="1"/>
  <SubFile name=".::book1.csv" num="2"/>
  <Attr checkSum="2928410085"
        csvQuote="&quot;"
        csvSeparate="\,"
        csvTerminate="\n,\r\n"
        fileCrc="-570320118"
        footerLength="0"
        format="csv"
        headerLength="0"
        job="book2.csv"
        maxRecordSize="8192"
        owner=""
        recordSize="102"
        size="204"
        workunit="D20150911-165823">
   <ECL>RECORD
    STRING field1;
    STRING field2;
    STRING field3;
END;</ECL>
  </Attr>
  <OrigName>.::superfile</OrigName>
 </SuperFile>
```